### PR TITLE
Add OC login copy commands

### PR DESCRIPTION
**Pull request by evidolob**
Demo:

https://user-images.githubusercontent.com/929743/232708809-01596bde-7bad-4e75-b52d-1ce09b873360.mov

PR also fixes https://github.com/crc-org/crc-extension/issues/46

Note: `Copy*` commands are shown only for `OpenShift` preset as `MicroShift` doesn’t provide necessary information

### CODE REVIEW

**Comment by cfergeau** [comment]
Fwiw, the ingress certificate error from `oc` should be fixed in https://mirror.openshift.com/pub/openshift-v4/x86_64/clients/ocp/candidate-4.12/

I don't know if we should expose `oc login` in the UI, or `oc config set-context`, the latter seems more in line with what podman desktop is doing ( https://podman-desktop.io/docs/kubernetes/kind/creating-a-kind-cluster - « In the Podman Desktop tray, open the Kubernetes menu, you can set the context to your Kind cluster: kind-<kind-cluster>. » )

File: src/extension.ts [hunk 1]
```diff
@@ -46,7 +47,7 @@ export async function activate(extensionContext: extensionApi.ExtensionContext):
 
   if (crcVersion) {
     status = 'installed';
-    connectToCrc();
+    await connectToCrc();
```
Comment (gbraad) [inline src/extension.ts:50]: :+1

File: src/login-commands.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import * as extensionApi from '@podman-desktop/api';
+
+import { commandManager } from './command';
+import { commander } from './daemon-commander';
+import { isWindows } from './util';
+
+let occommand = 'oc';
```
Comment (gbraad) [inline src/login-commands.ts:25]: we might need these commandnames somewhere less too, but for now, just leave them.

File: src/login-commands.ts [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+if (isWindows()) {
+  occommand = 'oc.exe';
+}
+
+export function registerLogInCommands(): void {
+  commandManager.addCommand({
+    id: 'crc.copy.login.admin',
+    label: 'Copy OC login command (admin)',
+    isEnabled: status => status.CrcStatus === 'Running',
```
Comment (gbraad) [inline src/login-commands.ts:34]: 👍

**Review by gbraad — APPROVED** [review]

**Comment by gbraad** [comment]
> I don't know if we should expose oc login in the UI,

You would suggest not to have these clipboard options?

> « In the Podman Desktop tray, open the Kubernetes menu, you can set the context to your Kind cluster: kind-. » )

That would be the better option...

... however, we also try to stay close to the current tray we are offering. When functionality is more rounded and complete, we could decide to drop these options, although a 'copy password' to clipboard might take the place for use in the console.

File: src/extension.ts [hunk 4]
```diff
@@ -156,13 +158,14 @@ export function deactivate(): void {
...
 async function registerOpenShiftLocalCluster(
+  name,
   provider: extensionApi.Provider,
   extensionContext: extensionApi.ExtensionContext,
 ): Promise<void> {
   const status = () => crcStatus.getConnectionStatus();
   const apiURL = 'https://api.crc.testing:6443';
   const kubernetesProviderConnection: extensionApi.KubernetesProviderConnection = {
-    name: 'OpenShift',
+    name,
```
Comment (gbraad) [inline src/extension.ts:168]: So this now is recognized as 'OpenShift Local', right?

Comment (evidolob) [reply src/extension.ts:168]: Yes https://github.com/crc-org/crc-extension/pull/48/files#diff-04bba6a35cad1c794cbbe677678a51de13441b7a6ee8592b7b50be1f05c6f626R222

**Pull request merged**